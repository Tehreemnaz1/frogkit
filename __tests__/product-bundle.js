import { mount, shallow } from 'vue-test-utils'
import Vue from 'vue'
import ProductBundle from '@/components/product-bundle'

const propsData = {
  product: {
    title: 'Kamera-Paket',
    bundle: [
      { 
        image: require('@demo/sf-app-icon.svg'),
        label: 'Smartfrog App'
      },
      { 
        image: require('@demo/sf-cam@2x.jpg'),
        label: 'Smartfrog HD Kamera'
      },
      { 
        image: require('@demo/sf-cloud.svg'),
        label: '4 Std. Video-Speicher'
      }
    ],
    features: [
      {
        placeholder: 'Live-Video rund um die Uhr'
      },
      {
        placeholder: 'Video-Speicherung',
        description: 'der jeweils letzten 4 Stunden'
      }
    ],
    additional: [
      'Kein Abo',
      'Keine Einrichtungskosten',
      'Zzgl. Versandkosten'
    ],
    link: {
      href: '#',
      text: 'Mehr Details'
    },
    price: {
      prefix: 'nur',
      amount: '5.95',
      currency: 'EUR',
      suffix: 'einmalg'
    },
    action: 'Jetzt kaufen'
  },
  hw: true,
  active: true
}


describe('Product Bundle component', () => {
  it('should render component and match snapshot', () => {
    // Note: a hack to get appropriate snapshot
    ProductBundle.data = () => { return {show: true}}
    const wrapper = mount(ProductBundle, {propsData})
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit click event', () => {
    const wrapper = mount(ProductBundle, {propsData})
    const stub = jest.fn()
    wrapper.vm.$on('click', stub)
    wrapper.findAll('button').at(0).trigger('click')
    expect(stub).toBeCalled()
  })
})
