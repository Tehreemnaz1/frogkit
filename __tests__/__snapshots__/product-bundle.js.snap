// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Bundle component should render component and match snapshot 1`] = `
<flex align="center" column="" class="fk-product-bundle fk-product-bundle__container">
    <title size="xs" color="primary" center="" class="fk-product-bundle__title">Kamera-Paket</title>
    <!---->
    <bundlerecap small="true" bundle="[object Object],[object Object],[object Object]"></bundlerecap>
    <div class="fk-product-bundle__features">
        <div class="fk-product-bundle__features__highlited">
            <div class="fk-product-bundle__features__container">
                <ticktext bold="" color="white" placeholder="Live-Video rund um die Uhr" class="fk-product-bundle__feature-item">
                    <!---->
                </ticktext>
                <ticktext bold="" color="white" placeholder="Video-Speicherung" class="fk-product-bundle__feature-item">
                    <p>der jeweils letzten 4 Stunden</p>
                </ticktext>
            </div>
        </div>
        <div class="fk-product-bundle__features__container">
            <ticktext placeholder="Kein Abo" class="fk-product-bundle__feature-item"></ticktext>
            <ticktext placeholder="Keine Einrichtungskosten" class="fk-product-bundle__feature-item"></ticktext>
            <ticktext placeholder="Zzgl. Versandkosten" class="fk-product-bundle__feature-item"></ticktext>
        </div>
    </div>
    <flex align="center" column="" class="fk-product-bundle__info"><a href="#" class="m-b-10">Mehr Details</a>
        <p class="m-b-15">
            <pricetag value="5.95" code="EUR" class="fk-product-bundle__price">
                <p slot="prefix" class="fk-product-bundle__price__prefix">nur</p>
                <p slot="suffix" class="fk-product-bundle__price__suffix">einmalg
                    <!---->
                </p>
            </pricetag>
        </p> <button color="secondary" size="big" block="" class="fk-product-bundle__action">Jetzt kaufen</button></flex>
</flex>
`;
